src/cmd/troubleshoot/image.go

```diff
@@ -98,92 +89,105 @@ func checkOneAgentCodeModulesImagePullable(troubleshootCtx *troubleshootContext)
+	pullSecret, err := getPullSecret(troubleshootCtx)
 	if err != nil {
 		return err
 	}
 
 	dynakubeActiveGateImage := getActiveGateImageEndpoint(troubleshootCtx)
+	err = checkComponentImagePullable(troubleshootCtx.httpClient, "ActiveGate", pullSecret, dynakubeActiveGateImage)
 
-	if err = checkComponentImagePullable(troubleshootCtx.httpClient, "ActiveGate", pullSecret, dynakubeActiveGateImage); err != nil {
+	if err != nil {
 		return err
 	}
 	return nil
 }
 
 func checkComponentImagePullable(httpClient *http.Client, componentName string, pullSecret string, componentImage string) error {
-	// split image path into registry and image nam
 	componentImageInfo, err := splitImageName(componentImage)
-	//	componentRegistry, componentImage, componentVersion, err := splitImageName(componentImage)
+
 	if err != nil {
 		return err
 	}
-	logInfof("using '%s' on '%s' with version '%s' as %s image", componentImageInfo.image, componentImageInfo.registry, componentImageInfo.version, componentName)
 
+	logInfof("using '%s' on '%s' with version '%s' as %s image", componentImageInfo.image, componentImageInfo.registry, componentImageInfo.version, componentName)
 	imageWorks := false
 
-	// parse docker config
 	var result Auths
-	if err := json.Unmarshal([]byte(pullSecret), &result); err != nil {
+	err = json.Unmarshal([]byte(pullSecret), &result)
+
+	if err != nil {
 		logErrorf("invalid pull secret, could not unmarshal to JSON: %v", err)
 		return nil
 	}
 
 	for registry, credentials := range result.Auths {
 		logInfof("checking images for registry '%s'", registry)
 
-		apiToken := base64.StdEncoding.EncodeToString([]byte(credentials.Username + ":" + credentials.Password))
+		err = registryAvailable(httpClient, registry+"/v2/", credentials.Auth)
 
-		if err := registryAvailable(httpClient, registry+"/v2/", apiToken); err != nil {
+		if err != nil {
 			logErrorf("%v", err)
 			continue
 		}
 
-		if err := imageAvailable(httpClient, "https://"+registry+"/v2/"+componentImageInfo.image+"/manifests/"+componentImageInfo.version, apiToken); err != nil {
-			logErrorf("cannot pull image '%s' with version '%s' from registry '%s': %v", componentImageInfo.image, componentImageInfo.version, registry, err)
-			continue
+		err = imageAvailable(httpClient, manifestUrl(registry, componentImageInfo), credentials.Auth)
+
+		if err != nil {
+			// Only print as a warning since other credentials might still work
+			// At this point it is uncertain if there is an error with the credentials
+			logWarningf("cannot pull image '%s' with version '%s' from registry '%s': %v",
+				componentImageInfo.image, componentImageInfo.version, registry, err)
 		} else {
-			logInfof("image '%s' with version '%s' exists on registry '%s", componentImageInfo.image, componentImageInfo.version, registry)
+			logInfof("image '%s' with version '%s' exists on registry '%s",
+				componentImageInfo.image, componentImageInfo.version, registry)
+			imageWorks = true
```

Comment:
couldn't we just log success and return here? no need for the `imageWorks` variable then - if the loop is completed the image was not available/found

Reply:
actually, change in logic here: before we would check all credentials, now we stop when the first one works. In my opinion stopping when the image is pullable is a perfectly fine approach